tests/test_camvid_format.py

```diff
@@ -0,0 +1,196 @@
+                            Mask(image=np.array([[1, 1, 0, 0, 0], [0, 0, 0, 0, 0]]), label=1),
+                            Mask(image=np.array([[0, 0, 1, 0, 0], [1, 1, 1, 1, 0]]), label=3),
+                            Mask(image=np.array([[0, 0, 0, 1, 1], [0, 0, 0, 0, 1]]), label=26),
+                        ]
+                    ),
+                    DatasetItem(id='0001TP_006690', subset='train',
+                        image=np.ones((1, 5, 3)),
+                        annotations=[
+                            Mask(image=np.array([[1, 1, 0, 1, 1]]), label=2),
+                            Mask(image=np.array([[0, 0, 1, 0, 0]]), label=17),
+                        ]
+                    ),
+                    DatasetItem(id='0016E5_07959', subset = 'val',
+                        image=np.ones((1, 5, 3)),
+                        annotations=[
+                            Mask(image=np.array([[1, 1, 1, 0, 0]]), label=0),
+                            Mask(image=np.array([[0, 0, 0, 1, 1]]), label=7),
+                        ]
+                    ),
+                ])
+
+        dataset = Project.import_from(DUMMY_DATASET_DIR, 'camvid').make_dataset()
+
+        compare_datasets(self, DstExtractor(), dataset)
+
+    def test_can_detect_camvid(self):
+        self.assertTrue(CamvidImporter.detect(DUMMY_DATASET_DIR))
+
+class CamvidConverterTest(TestCase):
+    def _test_save_and_load(self, source_dataset, converter, test_dir,
+            target_dataset=None, importer_args=None):
+        return test_save_and_load(self, source_dataset, converter, test_dir,
+            importer='camvid',
+            target_dataset=target_dataset, importer_args=importer_args)
+
+    def test_can_save_camvid_segm(self):
+        class TestExtractor(TestExtractorBase):
+            def __iter__(self):
+                return iter([
+                    DatasetItem(id='a/b/1', subset='test',
+                        image=np.ones((2, 5, 3)), annotations=[
+                        Mask(image=np.array([[0, 0, 0, 1, 0]]), label=0),
+                        Mask(image=np.array([[0, 1, 1, 0, 0]]), label=3),
+                        Mask(image=np.array([[1, 0, 0, 0, 1]]), label=4),
+                    ]),
+                ])
+
+        class DstExtractor(TestExtractorBase):
+            def __iter__(self):
+                return iter([
+                    DatasetItem(id='a/b/1', subset='test',
+                        image=np.ones((2, 5, 3)), annotations=[
+                        Mask(image=np.array([[0, 0, 0, 1, 0]]), label=0),
+                        Mask(image=np.array([[0, 1, 1, 0, 0]]), label=3),
+                        Mask(image=np.array([[1, 0, 0, 0, 1]]), label=4),
+                    ]),
+                ])
+
+        with TestDir() as test_dir:
+            self._test_save_and_load(TestExtractor(),
+                partial(CamvidConverter.convert, label_map='camvid'),
+                test_dir, target_dataset=DstExtractor())
```

Comment:
I can't see differences in the extractors above, I feel like a single definition is enough.